### PR TITLE
PR to support a more flexible camera naming.

### LINKED ISSUE #15: Make Camera Devices More General
## Expected Behavior
We can use any camera we want from the happi database. 

## Current Behavior
The code is rigidly expecting a camera name like `las_lhn_bay{x}_{n,f}f`. This makes it hard to give the cameras human-readable names. 

## Possible Solution
Make these names a configuration parameter rather than automatically determined. 

## Context
Some of our happi database names for cameras have been given more human readable names, which don't fit the current schema. This prevents us from being able to open these cameras.

### PR DESCRIPTION
**Pull request by slactjohnson**
## Description

Modify the btms to use a new property of the btms_config.SourcePosition class, introduced in this PR on pcdsdevices: https://github.com/pcdshub/pcdsdevices/pull/1265

## Motivation and Context
closes #15 

## How Has This Been Tested?
Tested interactively in the laser hall. 

## Where Has This Been Documented?
This PR.

### CODE REVIEW

**Comment by tangkong** [comment]
The version handling here broke for some reason, which is why the CI seems to be failing

**Comment by tangkong** [comment]
This might be related to the comment zach made in #9 about `_version.py` being auto-generated and not being needed to be tracked in the repository.

Our cookiecutter is setup to have a `version.py` file (no underscore) instead of `_version.py` (with underscore).  Can you try changing the name of that, and modifying `__init__.py` to match [the cookiecutter version?](https://github.com/pcdshub/cookiecutter-pcds-python/blob/master/%7B%7B%20cookiecutter.folder_name%20%7D%7D/%7B%7B%20cookiecutter.import_name%20%7D%7D/__init__.py)

**Comment by tangkong** [comment]
Playing around with this locally, I think this breaks because the tag format is not recognized by `setuptools_scm`.  The prepended `R` breaks the tag parsing.  

```python
> /cds/group/pcds/pyps/conda/py39/envs/pcds-5.8.4/lib/python3.9/site-packages/setuptools_scm/version.py(210)meta()
    208     node_date: date | None = None,
    209 ) -> ScmVersion:
--> 210     parsed_version = _parse_tag(tag, preformatted, config)
    211     log.info("version %s -> %s", tag, parsed_version)
    212     assert parsed_version is not None, "Can't parse version %s" % tag

ipdb> _parse_tag('1.0.0', preformatted, config)
<Version('1.0.0')>
```

Are we committed to `RX.X.X` as the tag format?  Or could we use `1.0.0`?

**Comment by slactjohnson** [comment]
If prefixing with 'R' is problematic, then I'm happy to change it. I just used the 'Rx.y.z' format because that's historically how we've tagged our IOC releases.

**Comment by tangkong** [comment]
Unfortunately the 'Rx.y.z' format isn't seen anywhere else from what I've seen.  Changing the release tag format is the best path forward for this python repo at least. 

For the purposes of this PR, we can ignore the CI and bypass the checks

**Comment by slactjohnson** [comment]
Thanks! I'll make this ready for review then and make a new tag without the 'R' after the merge.

**Comment by tangkong** [comment]
Requesting a review from @ZLLentz to double check my opinion on the tag / versioning issue

File: MANIFEST.in [hunk 1]
```diff
@@ -12,7 +12,7 @@ recursive-include btms_ui *.qss
 recursive-exclude btms_ui *.ui *.png
 
 include versioneer.py
-include btms_ui/_version.py
+include btms_ui/version.py
```
Comment (tangkong) [inline MANIFEST.in:15]:
```suggestion
include btms_ui/_version.py
```

I think this is actually still `_version.py` with the underscore.  The `_version.py` file gets autogenerated and packaged for pip / conda installs.

I will not claim this is obvious by any means though, I'm mostly comparing with our working packages 😆

File: conda-recipe/meta.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {% set package_name = "btms_ui" %}
 {% set import_name = "btms_ui" %}
-{% set version = load_file_regex(load_file=os.path.join(import_name, "_version.py"), regex_pattern=".*version = '(\S+)'").group(1) %}
+{% set version = load_file_regex(load_file=os.path.join(import_name, "version.py"), regex_pattern=".*version = '(\S+)'").group(1) %}
```
Comment (tangkong) [inline conda-recipe/meta.yaml:3]:
```suggestion
{% set version = load_file_regex(load_file=os.path.join(import_name, "_version.py"), regex_pattern=".*version = '(\S+)'").group(1) %}
```

Same as above, somewhat unintuitively

File: pyproject.toml [hunk 3]
```diff
@@ -23,7 +23,7 @@ file = "LICENSE.md"
 [project.scripts]
 
 [tool.setuptools_scm]
-write_to = "btms_ui/_version.py"
```
Comment (ZLLentz) [inline pyproject.toml:26]: This change should be undone

**Review by ZLLentz — COMMENTED** [review]
You should add this file exactly as-is: https://github.com/pcdshub/pcdsdevices/blob/master/pcdsdevices/version.py

To act as version proxy between installed/dev versions

**Comment by ZLLentz** [comment]
The core change here is :+1: just the packaging stuff

**Review by ZLLentz — APPROVED** [review]
I think everything is good here, if you merge and tag we can see if the tag build has similar issues or not

**Review by tangkong — DISMISSED** [review]
👍

**Comment by tangkong** [comment]
Builds after the new, compliant tag name look good.  Now only failing on a pypi deployment we haven't setup 🎉

**Pull request merged**